jetpack.php

```diff
@@ -93,6 +93,27 @@ function jetpack_admin_unsupported_wp_notice() { ?>
  */
 $jetpack_autoloader = JETPACK__PLUGIN_DIR . 'vendor/autoload_packages.php';
 if ( is_readable( $jetpack_autoloader ) ) {
+	add_filter( 'upgrader_post_install', 'jetpack_refresh_autoloader_post_install', 1, 2 );
+	/**
+	 * Refreshes the autoloader if the 'upgrader_post_install' hook fires. This
+	 * occurs when the plugin is updated.
+	 *
+	 * @param bool  $worked Installation response.
+	 * @param array $hook_extras Extra arguments passed to hooked filters.
+	 */
+	function jetpack_refresh_autoloader_post_install( $worked, $hook_extras ) {
+		if (
+			! isset( $hook_extras['plugin'] )
+			|| JETPACK__PLUGIN_FILE !== $hook_extras['plugin']
```

Comment:
Based on the description of the `$plugin` parameter [here](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/includes/class-plugin-upgrader.php#L138) and some testing, I don't think `JETPACK__PLUGIN_FILE` will work here. `JETPACK__PLUGIN_FILE` is the full path, but it looks like `$hook_extras['plugin']` will contain the plugin file path relative to the plugins directory.